### PR TITLE
chore: integrate ftl-provisioner with ftl serve

### PR DESCRIPTION
**Pull request by jvmakine**
Adds a hidden parameter to `ftl serve` to allow embedding provisioners to it. Uses this instead of a separate binary in the integration tests.

Next, I will work on a test for provisioning a mysql resource using the cloudformation plugin and localstack

### CODE REVIEW

File: frontend/cli/cmd_serve.go [hunk 1]
```diff
@@ -122,6 +134,13 @@ func (s *serveCmd) run(ctx context.Context, projConfig projectconfig.Config, ini
 		s.CommonConfig.AllowOrigins = append(s.CommonConfig.AllowOrigins, addr)
 	}
 
+	provisionerAddresses := make([]*url.URL, 0, s.Provisioners)
+	provisionerIngressAddresses := make([]*url.URL, 0, s.Provisioners)
+	for range s.Provisioners {
+		provisionerIngressAddresses = append(provisionerIngressAddresses, bindAllocator.Next())
```
Comment (alecthomas) [inline frontend/cli/cmd_serve.go:140]: What is "ingress" in the context of the provisioner? For the Controller that's the separate address for HTTP ingress requests, but that doesn't make much sense for the provisioner...

File: frontend/cli/cmd_serve.go [hunk 2]
```diff
@@ -404,7 +448,46 @@ func waitForControllerOnline(ctx context.Context, startupTimeout time.Duration,
 	}
 }
 
-func (s *serveCmd) isRunning(ctx context.Context, client ftlv1connect.ControllerServiceClient) bool {
+type pingableClient interface {
+	Ping(ctx context.Context, req *connect.Request[ftlv1.PingRequest]) (*connect.Response[ftlv1.PingResponse], error)
+}
+
+// waitForOnline polls qcclient of a service until it is online.
+func waitForOnline(ctx context.Context, name string, startupTimeout time.Duration, client pingableClient) error {
```
Comment (alecthomas) [inline frontend/cli/cmd_serve.go:456]: This function already exists [here](https://github.com/TBD54566975/ftl/blob/1ddd5830699bf866c66e79f3c2df20c898969117/internal/rpc/rpc.go#L149).

File: frontend/cli/cmd_serve.go [hunk 3]
```diff
@@ -404,7 +448,46 @@ func waitForControllerOnline(ctx context.Context, startupTimeout time.Duration,
 	}
 }
 
-func (s *serveCmd) isRunning(ctx context.Context, client ftlv1connect.ControllerServiceClient) bool {
+type pingableClient interface {
```
Comment (alecthomas) [inline frontend/cli/cmd_serve.go:451]: This interface already exists [here](https://github.com/TBD54566975/ftl/blob/1ddd5830699bf866c66e79f3c2df20c898969117/internal/rpc/rpc.go#L90).

**Review by alecthomas — APPROVED** [review]
Awesome!

**Comment by alecthomas** [comment]
Any thoughts on how to integrate plugins into the `ftl` binary too?

File: frontend/cli/cmd_serve.go (hunk 1 above)
Comment (jvmakine) [reply frontend/cli/cmd_serve.go:140]: Ah, this is my misunderstanding. Will be removed Pronto!

File: frontend/cli/cmd_serve.go (hunk 2 above)
Comment (jvmakine) [reply frontend/cli/cmd_serve.go:456]: Amazing!

**Comment by jvmakine** [comment]
> Any thoughts on how to integrate plugins into the ftl binary too?

Not yet, but that is the next thing I am looking at. I think I will just have a built-in `development` provisioner for now when running locally

**Pull request merged**